### ISSUE TITLE
Implement a Referer in requests to FreeIPA

### ISSUE THREAD

**Issue by n8rade** [camptocamp/terraform-provider-freeipa#28]
Unless something is misconfigured with my FreeIPA server and/or provider configuration I believe this provider will need to implement a Referer in requests due to this change: https://github.com/freeipa/freeipa/commit/13778d88ca2ac73b729821bdea844172a18c0cb9

When my FreeIPA updated to `ipa-server-4.9.12-11` I started to run into this error when using this provider:
~~~
╷
│ Error: Failed to connect to FreeIPA
│
│   with provider["registry.terraform.io/camptocamp/freeipa"],
│   on providers.tf line 28, in provider "freeipa":
│   28: provider "freeipa" {
│
│ Reason: initial login failed: unexpected http status code: 400
╵
~~~

These errors corresponded with the following error inside of `/var/log/httpd/error_log` on my FreeIPA node:
~~~
[Mon Jan 22 21:40:50.031433 2024] [wsgi:error] [pid 4726:tid 140282631108352] [remote 172.17.7.56:45794] ipa: ERROR: Rejecting request with missing Referer
[Mon Jan 22 21:40:55.555971 2024] [wsgi:error] [pid 4723:tid 140282631108352] [remote 172.17.7.56:36554] ipa: ERROR: Rejecting request with missing Referer
~~~

To get this working again I just ended up reverting part of the change on the FreeIPA side of this by changing this line to `True` instead of `False` and restarting services: https://github.com/freeipa/freeipa/blob/master/ipaserver/rpcserver.py#L162
Ideally I would've loved to have tested a change in the provider, but I'm very unfamiliar with Go and wasn't sure what I was looking at while I was poking around.

**Comment by marknl** [comment]
We ran into the same issue. For now we've changed the `rpcserver.py` as well so we can continue running our Terraform code.
Hoping so see an provider update that includes a Referer header soon! ;)

**Comment by yann-soubeyrand** [comment]
Should be fixed in [v0.8.1](https://github.com/camptocamp/terraform-provider-freeipa/releases/tag/v0.8.1).
